docs/introduction.md

```diff
@@ -67,83 +96,207 @@ Flow is based upon the following set of rules:
-Each chunk corresponds to a [collection](#collection-nodes).
-
-Errors introduced by the execution nodes are always guaranteed to have four critical attributes:
+Flow is designed to guarantee that any error introduced by the execution nodes are always guaranteed to have four critical attributes:
 
 * **Detectable**: A deterministic process has an objectively correct output. Therefore, even a single honest node in the network can detect deterministic faults and prove the error to all other honest nodes by pointing out the part of the process that was executed incorrectly.
 * **Attributable**: The output of all deterministic processes in Flow but be signed with the identity of the node that generated those results. As such, any error that has been detected can be clearly attributed to the node(s) that were responsible for that process.
 * **Punishable**: All nodes participating in a Flow network, including execution nodes, must put up a [stake](#staking) that can be [slashed](#slashing) if they are found to have exhibited [Byzantine behavior](#byzantine-fault). Since all errors in deterministic processes are detectable and attributable, those errors can be reliably punished via [slashing](#slashing).
 * **Recoverable**: The system must have a means to undo errors are they are detected. The property services to deter malicious actors from inducing errors that benefit them more than the slashing penalty.
 
-##### Consensus Nodes
+### Execution State vs Protocol State
+
+Flow differentiates between two sorts of states:
+1. Execution state (computational state)
+2. Protocol state (network infrastructure state)
+
+Execution and protocol states are maintained and updated independently. Execution state is maintained and updated by the execution nodes, while the protocol state is maintained and updated by the consensus nodes.
+
+The execution state contains the register values which are modified by the transaction execution. Updates to the execution states are done by the execution nodes but their integrity is governed by the verification process.
+
+The protocol state keeps track of the system related features like staked nodes, node roles, public keys, network addresses and staking amounts.
+It is updated when the nodes are slashed, join the system via staking or leave the system by unstaking.
+Consensus nodes publish updates on the protocol state directly as part of the blocks they generate.
+Consensus protocol guarantees the integrity of the protocol state.
+
+To better illustrate the difference and the independence of the two states, we can consider two scenarios:
+
+**Scenario 1**: Nodes never join, leave or change stake. Transactions are regularly processed. In this system, the protocol state never changes, but the execution state does.
+
+**Scenario 2**: No transactions are submitted/processed. Nodes are regularly joining or leaving the network — here only the network infrastructure/protocol state changes.
```

Comment:
Yep, that makes sense. However, please note that in scenario 2, the blockchain will still grow, even though the blocks are all empty. Please make sure to include the evolution of the block chain history as part of the protocol state.

Reply:
I touch on that in the `Block Formation` section:

> If there are no guaranteed collections, consensus nodes will continue block formation with an empty set of collections — block formation is never blocked.